### PR TITLE
Ensure that suppression state for build diagnostics shows up as activ…

### PR DESCRIPTION
**Pull request by mavasani**
…e instead of `<blank>`, and build diagnostics are not suppressible from error list, i.e. "Suppress" menu does not show up in the error list.

Fixes VSO [#164642](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems#_a=edit&id=164642)

### CODE REVIEW

**Comment by mavasani** [comment]
@srivatsn @heejaechang

File: src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs [hunk 1]
```diff
@@ -440,5 +444,21 @@ private static void GetLocationInfo(Document document, Location location, out Te
             originalLineInfo = location.GetLineSpan();
             mappedLineInfo = location.GetMappedLineSpan();
         }
+
+        /// <summary>
+        /// Properties for a diagnostic generated by an explicit build.
+        /// </summary>
+        internal static ImmutableDictionary<string, string> PropertiesForBuildDiagnostic => s_Properties;
```
Comment (heejaechang) [inline src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs]: why the indirection?

**Comment by heejaechang** [comment]
:+1:

File: src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs (hunk 1 above)
Comment (mavasani) [inline src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs]: Just moved it to a common place as now there are multiple clients that want to know if a diagnostic came from build.

**Pull request merged**